File: lib/java/src/main/java/uapi/UApiSchemaParseError.java
package uapi;

import static uapi.internal.schema.MapSchemaParseFailuresToPseudoJson.mapSchemaParseFailuresToPseudoJson;

import java.util.List;
import java.util.Map;

import uapi.internal.schema.SchemaParseFailure;

/**
 * Indicates failure to parse a uAPI Schema.
 */
public class UApiSchemaParseError extends RuntimeException {

    public final List<SchemaParseFailure> schemaParseFailures;
    public final List<Object> schemaParseFailuresPseudoJson;

    public UApiSchemaParseError(List<SchemaParseFailure> schemaParseFailures, Map<String, String> documentNamesToJson) {
        super(String.valueOf(mapSchemaParseFailuresToPseudoJson(schemaParseFailures, documentNamesToJson)));
        this.schemaParseFailures = schemaParseFailures;
        this.schemaParseFailuresPseudoJson = mapSchemaParseFailuresToPseudoJson(schemaParseFailures,
                documentNamesToJson);
    }

    public UApiSchemaParseError(List<SchemaParseFailure> schemaParseFailures, Map<String, String> documentNamesToJson,
            Throwable cause) {
        super(String.valueOf(mapSchemaParseFailuresToPseudoJson(schemaParseFailures, documentNamesToJson)), cause);
        this.schemaParseFailures = schemaParseFailures;
        this.schemaParseFailuresPseudoJson = mapSchemaParseFailuresToPseudoJson(schemaParseFailures,
                documentNamesToJson);
    }
}

File: lib/ts/uapi/UApiSchemaParseError.ts
import { SchemaParseFailure } from 'uapi/internal/schema/SchemaParseFailure';
import { mapSchemaParseFailuresToPseudoJson } from 'uapi/internal/schema/MapSchemaParseFailuresToPseudoJson';

export class UApiSchemaParseError extends Error {
    /**
     * Indicates failure to parse a uAPI Schema.
     */

    public schemaParseFailures: SchemaParseFailure[];
    public schemaParseFailuresPseudoJson: any;

    constructor(schemaParseFailures: SchemaParseFailure[], cause?: Error) {
        super(mapSchemaParseFailuresToPseudoJson(schemaParseFailures).toString(), { cause: cause });
        this.schemaParseFailures = schemaParseFailures;
        this.schemaParseFailuresPseudoJson = mapSchemaParseFailuresToPseudoJson(schemaParseFailures);
    }
}
